products/bash/configure-apic-v10.sh

```diff
@@ -83,6 +83,17 @@ if [[ "$STAGING_AUTHS" == "" || "$STAGING_AUTHS" == "null" ]]; then
 else
   REPO="cp.stg.icr.io"
 fi
+
+if [[ $(oc get secret cp4i-demo-apic-smtp-secret -n "$NAMESPACE") ]]; then
+  MAIL_SERVER_HOST=$(oc get secret cp4i-demo-apic-smtp-secret -n "$NAMESPACE" -o json | jq -r '.data.mailServerHost' | base64 --decode)
+  MAIL_SERVER_PORT=$(oc get secret cp4i-demo-apic-smtp-secret -n "$NAMESPACE" -o json | jq -r '.data.mailServerPort' | base64 --decode)
+  MAIL_SERVER_USERNAME=$(oc get secret cp4i-demo-apic-smtp-secret -n "$NAMESPACE" -o json | jq -r '.data.username' | base64 --decode)
+  MAIL_SERVER_PASSWORD=$(oc get secret cp4i-demo-apic-smtp-secret -n "$NAMESPACE" -o json | jq -r '.data.password' | base64 --decode)
```

Comment:
I think we should use the same names as originally used in the CR:
```suggestion
  MAIL_SERVER_USERNAME=$(oc get secret cp4i-demo-apic-smtp-secret -n "$NAMESPACE" -o json | jq -r '.data.mailServerUsername' | base64 --decode)
  MAIL_SERVER_PASSWORD=$(oc get secret cp4i-demo-apic-smtp-secret -n "$NAMESPACE" -o json | jq -r '.data.mailServerPassword' | base64 --decode)
```

Reply:
When creating a secret without giving a `username` and `password` in it, it gives an error saying `username` and `password` are required fields:
```bash
The Secret "cp4i-demo-apic-smtp-secret-1" is invalid: 
* data[%s][username]: Required value
* data[%s][password]: Required value
```

</br>

The secret yaml:
```yaml
cat <<EOF | oc apply --namespace ${NAMESPACE} -f -
apiVersion: v1
kind: Secret
metadata:
  name: cp4i-demo-apic-smtp-secret-1
type: kubernetes.io/basic-auth
stringData:
  emailAddress: "your@email.address"
  mailServerHost: "smtp.mailtrap.io"
  mailServerPort: "2525"
  mailServerUsername: "your-username"
  mailServerPassword: "your-password123"
EOF
```

Reply:
The type should be:
```
type: Opaque
```
See [these docs](https://kubernetes.io/docs/concepts/configuration/secret/#secret-types)

![image](https://user-images.githubusercontent.com/64848511/113691880-d5397100-96c4-11eb-942b-dd9ba03c4130.png)

When specifying basic auth as the type you need username/password. For Opaque you can have any you like.

Reply:
Done